### PR TITLE
virtcontainers: remove sandboxConfig.VMConfig

### PR DESCRIPTION
**Pull request by bergwolf**
We can just use hyprvisor config to specify the memory size
of a guest. There is no need to maintain the extra place just
for memory size.

Fixes: #692

### CODE REVIEW

File: virtcontainers/pkg/oci/utils.go [hunk 1]
```diff
@@ -406,25 +404,23 @@ func (spec *CompatOCISpec) SandboxID() (string, error) {
...
 		ocispec.Linux.Resources.Memory.Limit != nil {
 		memBytes := *ocispec.Linux.Resources.Memory.Limit
 		if memBytes <= 0 {
-			return vc.Resources{}, fmt.Errorf("Invalid OCI memory limit %d", memBytes)
+			return fmt.Errorf("Invalid OCI memory limit %d", memBytes)
 		}
 		// Use some math magic to round up to the nearest Mb.
 		// This has the side effect that we can never have <1Mb assigned.
-		resources.Memory = uint((memBytes + (1024*1024 - 1)) / (1024 * 1024))
+		config.HypervisorConfig.DefaultMemSz = uint32((memBytes + (1024*1024 - 1)) / (1024 * 1024))
```
Comment (amshinde) [inline virtcontainers/pkg/oci/utils.go]: @bergwolf I think we should rename this to not include "Default" anymore, considering how this var is being used now.

**Comment by bergwolf** [comment]
updated. @amshinde @sboeuf PTAL.

**Comment by jcvenegas** [comment]
\o/ nice PR, @bergwolf  could you update the commit to allow the CI test the PR.
```
ERROR: Commit 70973499bcdd5954632309a8daa1e0170d820861: Failed to find subsystem in subject: "rename DefaultVCPUs and DefaultMemSz"
ERROR: checkcommits failed. See the document below for help on formatting
commits for the project.
```

**Comment by jodh-intel** [comment]
Hi @bergwolf - tests are failing:

```
=== RUN   TestSandboxStatePausedReady
time="2018-09-06T03:04:50Z" level=error msg="trace called before context set" sandbox=100 source=virtcontainers subsystem=sandbox type=bug
```

It looks like `Sandbox.ctx == nil` for some reason?

**Comment by jodh-intel** [comment]
@sboeuf, @amshinde - are you happy with this now?

**Pull request merged**